cloud_component/main.py

```diff
@@ -16,31 +18,78 @@
 import proto.fog_pb2_grpc as fog_pb2_grpc  # noqa
 
 class Task:
-    def __init__(self, sensor_data):
-        self.id = uuid4()
-        self.value = sensor_data.value
-        self.result = None
-        self.processed = False
+    def __init__(self,value, task_id=None, result= None, processed= False):
+        if task_id is None:
+            task_id=uuid4()
+        self.id = task_id
+        self.value = value
+        self.result = result
+        self.processed = processed
+
+    def to_dict(self):
+        return {
+            "id": str(self.id),
+            "value": self.value,
+            "result": self.result,
+            "processed": self.processed
+        }
+
+    @classmethod
+    def from_dict(cls, data):
+        return cls(value=data["value"], task_id=data["id"], result=data["result"], processed=data["processed"])
 
     def __str__(self):
         return f'Task_id: {self.id}, processed: {self.processed}'
     
 class TaskQueue:
     def __init__(self):
-        self.tasks = queue.Queue()
+        self.tasks = deque()
         self.lock = threading.Lock()
-
+        self.persistence_file_path = os.path.join(os.path.dirname(__file__), "queue_replica.json")
+
+        self.load_tasks_from_replica_queue()
+
+    def load_tasks_from_replica_queue(self):
+        if os.path.exists(self.persistence_file_path):
+            with open(self.persistence_file_path, "r") as f:
+                backup_data = load(f)
+                for task in backup_data:
+                    instance_task = Task.from_dict(task)
+                    self.add_task(task=instance_task)
+
+    def write_tasks_to_replica_queue(self):
+        if os.path.exists(self.persistence_file_path):
```

Comment:
Nit: I had to manually create the file, otherwise nothing is persisted. Perhaps we should create it when the service starts up (if it doesn't already exist).

Reply:
resolved in 34c3ffefc250d3ae97b9737696f964ec9c140dab